config/peerpods/podvm/libvirt-config-manager.sh

```diff
@@ -0,0 +1,380 @@
+data:
+  CLOUD_PROVIDER: "$CLOUD_PROVIDER"
+  PROXY_TIMEOUT: "15m"
+EOF
+)
+
+install_sshpass() {
+    if ! command -v sshpass &> /dev/null; then
+        echo "Installing sshpass..."
+         dnf install -y sshpass
+    fi
+}
+
+# Function to generate SSH keys and copy them to the KVM host
+# this is required to establish connection with KVM host for managing libvirt resources
+generate_ssh_keys() {
+    ssh-keygen -t rsa -b 4096 -f /root/.ssh/id_rsa \
+        -N "${SSH_KEY_PASSPHRASE}" >/dev/null 2>&1
+    
+    sshpass -p "${KVM_HOST_PASSWORD}" ssh-copy-id \
+        -o StrictHostKeyChecking=no "${KVM_HOST_USERNAME}@${KVM_HOST_ADDRESS}" \
+        >/dev/null 2>&1
+
+    oc create secret generic ssh-key-secret \
+        -n openshift-sandboxed-containers-operator \
+        --from-file=id_rsa.pub=/root/.ssh/id_rsa.pub \
+        --from-file=id_rsa=/root/.ssh/id_rsa
+
+    echo "SSH keypair generated and copied to ${KVM_HOST_ADDRESS}, secret created with pub & private keys."
+}
+
+# Function to create a libvirt pool and volume on the KVM host, and sync SSH keys
+create_pool_volume_on_kvm_and_sync_sshid() {
+    local kvm_host_user="$1"
+    local kvm_host_address="$2"
+    local libvirt_pool="$3"
+    local libvirt_vol_name="$4"
+    local libvirt_pool_directory="$5"
+
+    echo "Creating pool and volume on KVM host '${kvm_host_address}'..."
+    ssh -i /root/.ssh/id_rsa -o StrictHostKeyChecking=no \
+        "${KVM_HOST_USERNAME}@${kvm_host_address}" << EOF
+        sudo mkdir "$libvirt_pool_directory"
+        sudo virsh pool-define-as "${libvirt_pool}" --type dir \
+            --target "$libvirt_pool_directory"
+        sudo virsh pool-start "${libvirt_pool}"
+        
+        sudo virsh -c qemu:///system vol-create-as --pool "${libvirt_pool}" \
+            --name "${libvirt_vol_name}" --capacity 20G --allocation 2G \
+            --prealloc-metadata --format qcow2
+        
+        sudo cat /home/$kvm_host_user/.ssh/authorized_keys | \
+            sudo tee -a /root/.ssh/authorized_keys > /dev/null 2>&1
+EOF
+}
+
+# checking if the libvirt pool and volume already exist
+check_pool_and_volume_existence() {
+    local kvm_host_address="$1"
+    local libvirt_pool="$2"
+    local libvirt_vol_name="$3"
+
+    echo "Checking existence of libvirt pool '${libvirt_pool}' and volume '${libvirt_vol_name}' on KVM host '${kvm_host_address}'..."
+    ssh -i /root/.ssh/id_rsa -o StrictHostKeyChecking=no \
+        "${KVM_HOST_USERNAME}@${kvm_host_address}" << EOF
+        sudo virsh pool-info "${libvirt_pool}" >/dev/null 2>&1
+        POOL_EXISTS=\$?
+        sudo virsh vol-info --pool "${libvirt_pool}" "${libvirt_vol_name}" \
+            >/dev/null 2>&1
+        VOL_EXISTS=\$?
+
+        if [ "\$POOL_EXISTS" -eq 0 ] && [ "\$VOL_EXISTS" -eq 0 ]; then
+            echo "A Libvirt pool named '${libvirt_pool}' with volume '${libvirt_vol_name}' already exists on the KVM host. Please choose a different name."
+            exit 0
```

Comment:
There is no way to re-use an existing pool?
No big deal if it's a mandatory requirement, but if we can keep a little bit of flexibility, that would be nice - if only for development/test setups :-)